src/transformers/trainer_utils.py

```diff
@@ -46,7 +46,27 @@
     import tensorflow as tf
 
 
-def set_seed(seed: int):
+def seed_worker():
+    worker_seed = torch.initial_seed() % 2**32
+    set_seed(worker_seed)
+
+
+def set_deterministic_seed_for_cuda():
+    #  Enable PyTorch deterministic mode. This potentially requires either the environment
+    #  variable 'CUDA_LAUNCH_BLOCKING' or 'CUBLAS_WORKSPACE_CONFIG' to be set,
+    # depending on the CUDA version, so we set them both here
+    os.environ["CUDA_LAUNCH_BLOCKING"] = "1"
+    os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":16:8"
```

Comment:
These are only relevant if torch is available, so I'd move them down into the if-block.